From: Diego Escalante Urrelo <diegoe@gnome.org>
Date: Fri, 14 Feb 2025 00:06:29 -0500
Subject: wl: Fix memcpy() field-spanning write warning

Handle the following warning:
  kernel: memcpy: detected field-spanning write (size 160) of single
  field "dst" at .../src/wl/sys/wl_cfg80211_hybrid.c:3086 (size 0)

See: https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=1084853
---
 src/wl/sys/wl_cfg80211_hybrid.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git src/wl/sys/wl_cfg80211_hybrid.h src/wl/sys/wl_cfg80211_hybrid.h
index bc6f3ad..014c399 100644
--- a/src/wl/sys/wl_cfg80211_hybrid.h
+++ b/src/wl/sys/wl_cfg80211_hybrid.h
@@ -103,7 +103,7 @@ struct beacon_proberesp {
 	__le64 timestamp;
 	__le16 beacon_int;
 	__le16 capab_info;
-	u8 variable[0];
+	u8 variable[];
 } __attribute__ ((packed));
 
 struct wl_cfg80211_conf {
